misc/ls-extensions/modules/performance-analyzer-services/src/main/java/io/ballerina/PerformanceAnalyzerNodeVisitor.java

```diff
@@ -0,0 +1,536 @@
+            ExpressionNode expNode = ((CheckExpressionNode) expressionNode).expression();
+            resolveEndPoint(symbol, expNode);
+            return;
+        } else {
+            return;
+        }
+
+        TypeSymbol typeSymbol = getTypeDescriptor(((VariableSymbol) symbol).typeDescriptor());
+
+        String signature = typeSymbol.signature();
+        Optional<ModuleSymbol> typeSymbolModule = typeSymbol.getModule();
+        if (typeSymbolModule.isEmpty() || symbol.getLocation().isEmpty()) {
+            return;
+        }
+
+        LineRange lineRange = symbol.getLocation().get().lineRange();
+        EndPointNode endpoint = new EndPointNode(typeSymbolModule.get().id().orgName() +
+                "/" + typeSymbolModule.get().id().moduleName(),
+                signature.substring(signature.lastIndexOf(":") + 1), url, lineRange);
+        this.endPointDeclarationMap.put(getUUID(lineRange), endpoint);
+    }
+
+    private String getUUID(LineRange lineRange) {
+
+        if (referenceMap.containsKey(lineRange)) {
+            return referenceMap.get(lineRange);
+        }
+
+        String uuid = UUID.randomUUID().toString();
+        referenceMap.put(lineRange, uuid);
+        return uuid;
+    }
+
+    private String findBaseUrl(SeparatedNodeList<FunctionArgumentNode> arguments) {
+        // Assuming that the first argument is the base URL.
+        for (FunctionArgumentNode argumentNode : arguments) {
+            UrlFinder urlFinder = new UrlFinder();
+            argumentNode.accept(urlFinder);
+
+            String url = urlFinder.getToken();
+            if (url.isEmpty()) {
+                continue;
+            }
+            if (argumentNode.kind() == SyntaxKind.POSITIONAL_ARG &&
+                    ((PositionalArgumentNode) argumentNode).expression().kind() == SyntaxKind.STRING_LITERAL) {
+                argumentNode.accept(this);
+                return extractURL(url);
+            } else if (argumentNode.kind() == SyntaxKind.NAMED_ARG &&
+                    ((NamedArgumentNode) argumentNode).expression().kind() == SyntaxKind.STRING_LITERAL) {
+                argumentNode.accept(this);
+                return extractURL(url);
+            }
+        }
+        return null;
+    }
+
+    private String extractURL(String arg) {
+
+        if (arg.startsWith("\"") && arg.endsWith("\"")) {
+            return arg.substring(1, arg.length() - 1);
+        } else {
+            return arg;
+        }
+    }
+
+    private boolean isClientObject(Symbol symbol) {
+
+        if (symbol.kind() != SymbolKind.VARIABLE) {
+            return false;
+        }
+
+        TypeSymbol typeSymbol = getTypeDescriptor(((VariableSymbol) symbol).typeDescriptor());
+
+        return typeSymbol instanceof Qualifiable && ((Qualifiable) typeSymbol).qualifiers().contains(Qualifier.CLIENT);
+    }
+
+    private boolean isRecordObject(Symbol symbol) {
+
+        return getTypeKind(symbol) == TypeDescKind.RECORD;
```

Comment:
Are you referring to the type descriptors here? If so does this support TypeReferenceTypeSymbols as well?